### ISSUE TITLE
docker.service cookbook template conflicting with docker.service parameters from docker-engine 17.04

### ISSUE THREAD

**Issue by mlcooper** [chef-cookbooks/docker#868]
### Cookbook version
`2.15.6`

### Chef-client version
`12.14.89`

### Platform Details
Red Hat Enterprise Linux Server release 7.3 (Maipo)
VM running in VMware
kernel `3.10.0-514.21.1.el7.x86_64`

### Scenario:
I am trying to upgrade to docker `17.04.0.ce-1.el7.centos` from `17.03.0.ce-1.el7.centos` due to https://github.com/moby/moby/issues/33615, which changes a very important config item `After` in the file `/etc/systemd/system/docker.service`.  

The problem is that when I am using this docker cookbook to do that, the template you have in this cookbook overrides the new settings that docker put in `/etc/systemd/system/docker.service` in 17.04 version of the docker-engine.  

This is the `/etc/systemd/system/docker.service` from docker in 17.04, https://github.com/moby/moby/blob/v17.04.0-ce/contrib/init/systemd/docker.service, but as you can see in your template file here, specifically this line, https://github.com/chef-cookbooks/docker/blob/master/templates/default/systemd/docker.service.erb#L4, overrides what docker put in there, so I'm not able to fix my issue I'm having by upgrading to 17.04.

### Steps to Reproduce:
Use this docker cookbook on a RHEL7 box to install docker 17.03.  Then change the `package_version` option in the `docker_service` resource to install `docker-engine 17.04.0.ce-1.el7.centos`, and you will see that your `/etc/systemd/system/docker.service` file does not, and I think cannot, match the default values docker puts in this file, shown here:  https://github.com/moby/moby/blob/v17.04.0-ce/contrib/init/systemd/docker.service

### Expected Result:
I expect that when I use the `docker_service` resource to install docker, I will by default get what docker wants in https://github.com/moby/moby/blob/v17.04.0-ce/contrib/init/systemd/docker.service, without having to really override it with something like the `systemd_opts` option in the `docker_service` resource, unless that's your intention.  If it is, then every time someone upgrades `docker-engine`, they'd have to manually ensure that the parameters in their `docker.service` file are in fact what docker wants in there with the version they just upgraded to.

### Actual Result:
What actually happens after I upgrade to 17.04, my docker.service file does not have the new parameters docker put in the `After` line.  It still looks like this:
```
[root@foo system]# cat /etc/systemd/system/docker.service
[Unit]
Description=Docker Application Container Engine
Documentation=http://docs.docker.com
After=network.target
```
I tried to use the `syslogd_opts` option for the `docker_service` resource, but it puts the option under the `[Service]` section of the `docker.service` file, and the setting I need to change is the `After` setting under the `[Unit]` section.  With the templates in this cookbook, I don't see a way to affect any change in the `[Unit]` section.

Diverging of the template in this cookbook from the docker service file is also mentioned in issue https://github.com/chef-cookbooks/docker/issues/840

**Comment by mlcooper** [comment]
As a workaround wrapped the docker cookbook and put an override file here:
`/etc/systemd/system/docker.service.d/override.conf`, with this content:
```
[Unit]
After=
After=network-online.target firewalld.service
```
And this seems to be working so far.

**Comment by chasebolt** [comment]
resolved in release 2.15.9 (49064e1f282b95e53055fdf996f2a7d9d60a2e7f)
